scripts/populate_tox/populate_tox.py

```diff
@@ -0,0 +1,502 @@
+        if classifier.startswith(CLASSIFIER_PREFIX):
+            python_version = classifier[len(CLASSIFIER_PREFIX) :]
+            if "." in python_version:
+                # We don't care about stuff like
+                # Programming Language :: Python :: 3 :: Only,
+                # Programming Language :: Python :: 3,
+                # etc., we're only interested in specific versions, like 3.13
+                python_versions.append(Version(python_version))
+
+    if python_versions:
+        python_versions.sort()
+        return python_versions
+
+    # We only use `requires_python` if there are no classifiers. This is because
+    # `requires_python` doesn't tell us anything about the upper bound, which
+    # depends on when the release first came out
+    try:
+        requires_python = pypi_data["info"]["requires_python"]
+    except (AttributeError, KeyError):
+        pass
+
+    if requires_python:
+        return SpecifierSet(requires_python)
+
+    return []
+
+
+def _render_python_versions(python_versions: list[Version]) -> str:
+    return (
+        "{"
+        + ",".join(f"py{version.major}.{version.minor}" for version in python_versions)
+        + "}"
+    )
+
+
+def _render_dependencies(integration: str, releases: list[Version]) -> list[str]:
+    rendered = []
+
+    if TEST_SUITE_CONFIG[integration].get("deps") is None:
+        return rendered
+
+    for constraint, deps in TEST_SUITE_CONFIG[integration]["deps"].items():
+        if constraint == "*":
+            for dep in deps:
+                rendered.append(f"{integration}: {dep}")
+        elif constraint.startswith("py3"):
+            for dep in deps:
+                rendered.append(f"{constraint}-{integration}: {dep}")
+        else:
+            restriction = SpecifierSet(constraint)
+            for release in releases:
+                if release in restriction:
+                    for dep in deps:
+                        rendered.append(f"{integration}-v{release}: {dep}")
+
+    return rendered
+
+
+def write_tox_file(packages: dict) -> None:
+    template = ENV.get_template("tox.jinja")
+
+    context = {"groups": {}}
+    for group, integrations in packages.items():
+        context["groups"][group] = []
+        for integration in integrations:
+            context["groups"][group].append(
+                {
+                    "name": integration["name"],
+                    "package": integration["package"],
+                    "extra": integration["extra"],
+                    "releases": integration["releases"],
+                    "dependencies": _render_dependencies(
+                        integration["name"], integration["releases"]
+                    ),
+                }
+            )
+
+    rendered = template.render(context)
+
+    with open(TOX_FILE, "w") as file:
+        file.write(rendered)
+        file.write("\n")
+
+
+def _get_package_name(integration: str) -> Tuple[str, Optional[str]]:
```

Comment:
```suggestion
def _get_package_name(integration: str) -> tuple[str, Optional[str]]:
```

I think the builtin `tuple` support subscripting